app/services/reports/base_report.rb

```diff
@@ -4,6 +4,15 @@ module Reports
   class BaseReport
     private
 
+    def fiscal_start_date
+      now = Time.zone.now
+      if now.strftime('%m').to_i >= 10
+        now.strftime('10-01-%Y')
+      else
+        "10-01-#{now.strftime('%Y').to_i - 1}"
+      end
+    end
+
     def first_of_this_month
       Time.zone.today.strftime('%m-01-%Y')
```

Comment:
Rails has a helper for this:
https://apidock.com/rails/ActiveSupport/CoreExtensions/Date/Calculations/beginning_of_month

```ruby
Time.zone.today.beginning_of_month
```

and then the `.to_s` is the `YYYY-MM-DD` already so:

```ruby
Time.zone.today.beginning_of_month.to_s
```